src/main/java/org/fenixedu/academic/domain/accounting/DueDateAmountMap.java

```diff
@@ -142,4 +142,17 @@ public void clear() {
     private static <T> T throwImmutable() {
         throw new UnsupportedOperationException("This instance is immutable");
     }
+
+    @Override
+    public boolean equals(final Object obj) {
+        if (!(obj instanceof DueDateAmountMap)) {
```

Comment:
Why not just do this instead:
`return obj instanceof DueDateAmountMap && innerMap.equals(obj);`